components/authz-service/server/v2/project_update_manager.go

```diff
@@ -432,3 +404,45 @@ func (m *CerealProjectUpdateManager) Status() (ProjectUpdateStatus, error) {
 	}
 	return projectUpdateInstance, nil
 }
+
+// Find the longest estimate that is in the future. If no estimate is found return zero time.Time{}
+func findLongestEstimatedTime(estimatedTimes []time.Time) time.Time {
+	longestEstimatedTimeComplete := time.Time{}
+	for _, estimatedTime := range estimatedTimes {
+		if estimatedTime.After(time.Now()) && estimatedTime.After(longestEstimatedTimeComplete) {
+			longestEstimatedTimeComplete = estimatedTime
+		}
+	}
+
+	return longestEstimatedTimeComplete
+}
+
+func (w *workflowInstance) collectAllDomainServiceEstimates() []time.Time {
+	estimates := make([]time.Time, 0)
+	if !w.IsRunning() {
+		return estimates
+	}
+	domainServicesUpdateInstance, err := w.GetUpdateDomainServicesInstance()
+	if err == cereal.ErrWorkflowInstanceNotFound {
+		return estimates
+	}
```

Comment:
❓ Can there be other errors that we might care about? -- Can `err` be not `nil` and not `cereal.ErrWorkflowInstanceNotFound`?

Reply:
I am not sure. Maybe @jaym might know this question.

Reply:
i think not. but it doesn't hurt to defend against it